modules/core/src/main/scala/fs2/kafka/CommittableProducerRecords.scala

```diff
@@ -67,8 +63,17 @@ object CommittableProducerRecords {
   def apply[F[_], G[+_], K, V](
     records: G[ProducerRecord[K, V]],
     offset: CommittableOffset[F]
-  )(implicit G: Foldable[G]): CommittableProducerRecords[F, G, K, V] =
-    new CommittableProducerRecordsImpl(records, offset, G)
+  )(implicit G: Foldable[G]): CommittableProducerRecords[F, K, V] = {
+    val buf = new mutable.ArrayBuffer[ProducerRecord[K, V]]()
```

Comment:
I think it's worth initializing this to `records.size`, as many `Foldable` have constant-time `size`.

Reply:
We can also specialize with `Chunk.empty` and `Chunk.singleton` on sizes 0 and 1, respectively. Especially since `one` uses `apply` (also possible to change `one` to `new CommittableProducerRecordsImpl(Chunk.singleton(record), offset)`).

Reply:
`size` on `Foldable` returns `Long`, but `ArrayBuffer`'s constructor expects `Int`. What's the best way to safely handle the overflow?

Reply:
I'd say it's unlikely you would want to produce more than `Int.MaxValue` records at a time, so `.toInt` sounds reasonable to me.